src/main/java/com/sphy/hotelmanagementapplication/service/HotelService.java

```diff
@@ -105,35 +145,59 @@ public HotelDTO updateHotel(HotelDTO hotelDTO, List<RoomDTO> roomDTOS) throws Nu
 			existingHotel.setAreaName(hotelDTO.getAreaName());
 			Optional<Admin>  admin = adminRepository.findById(hotelDTO.getId());
 			admin.ifPresent(existingHotel::setOwner);
-			for (RoomDTO roomDTO : roomDTOS){
-				roomService.updateRoom(roomDTO);
-			}
+
 			hotelRepository.save(existingHotel);
 		}
 		return hotelDTO;
 	}
 
+	/***
+	 * saves a hotel
+	 * @param hotelDTO hotel to be saved
+	 * @return the saved hotel for confirmation
+	 * @throws Exception
+	 */
 	public HotelDTO saveHotelDTO(HotelDTO hotelDTO) throws Exception {
-		Hotel hotel = new Hotel();
+		Hotel hotel = new Hotel(1L);
 		Optional<Admin> adminOpt =
 				adminRepository.findById(hotelDTO.getOwner());
-		adminOpt.ifPresent(hotel::setOwner);
-		hotel.setName(hotelDTO.getName());
-		hotel.setAreaName(hotelDTO.getAreaName());
-		hotel.setStars(hotelDTO.getStars());
 
-		return hotelDTO;
+		Set<RoomDTO> roomOpt = hotelDTO.getRooms();
+
+		if (adminOpt.isPresent() && roomOpt.size()>0){
+			hotel = hotelDTOToHotel.converter(hotelDTO);
+		}
+		if (!adminOpt.isPresent()){
+			throw new Exception("There is no Owner registered with that id, or the id is null!");
+		}
+		if (roomOpt.size() == 0){
+			throw new Exception("If you wont to save hotels you mast add one or more rooms");
+		}
+
+		return hotelToHotelDTO.converter(hotel);
 	}
 
-	public List<HotelDTO> saveHotels(List<HotelDTO> hotelsDTO, List<RoomDTO> roomsDTOS) throws Exception {
+	/***
+	 * save a list of hotels
+	 * @param hotelsDTO list of hotels to be saved
+	 * @return the saved hotels for confirmation
+	 * @throws Exception
+	 */
+	public List<HotelDTO> saveHotels(List<HotelDTO> hotelsDTO) throws Exception {
 		List<Hotel> hotels = new ArrayList<>();
-
 		for (HotelDTO hotelDTO : hotelsDTO){
+
+			if (hotelDTO.getOwner() == null || adminService.getAdminById(hotelDTO.getOwner()) == null) {
+				throw new Exception("There is no Owner registered with that id, or the id is null!");
+			}
+
+			if (hotelDTO.getRooms() == null){
+				throw new Exception("If you wont to save hotels you mast add one or more rooms");
+			}
+
 			hotels.add(hotelDTOToHotel.converter(hotelDTO));
 		}
 
-		roomService.saveRooms(roomsDTOS);
-
 		hotelRepository.saveAll(hotels);
```

Comment:
The saveAll returns an Iterable of the objects that were saved. Please return those after converting them to DTOS